### ISSUE TITLE
Crashes on old android version

### ISSUE THREAD

**Issue by rVicio** [Waterdish/Shipwright-Android#13]
Hi, I want to inform about a crash problem in older Android versions:
I'm using an old samsung galaxy J2 prime with Android 6.0.1, the soh.apk installs with no problems but it crashes when opening it with both otr files (oot.otr and soh.otr) generated on pc. I tried deleting the otr files, reinstalling the apk and let it generate the otr files itself, it shows the "extract" message and it actually finds the rom, when I select to extract, it stays in black screen for about 3 minutes and then ask if i want to extract another rom, just as stated in the instructions, but crashes after selecting "no" to start, I checked the assets folder and it generates the otr files just fine but still crashes.
Also, just for curiosity, I tested the soh.apk in an samsung galaxy grand neo plus (android 4.4.4), which only has up to OpenGL ES 2.0, expecting it to crash due to no supporting the minimun graphics API, but the crash message says it can't find something related to SDL2.

**Comment by Waterdish** [comment]
Thank you. I'll see if I can find out what might be causing those crashes.

**Comment by rVicio** [comment]
Thanks for the quick answer, there's one more thing I want to add:
The oot.otr file generated in the android 6.0.1 doesn't work even with the pc .exe, but the soh.otr generated on android works just fine with the oot.otr file generated on PC. Basically, the extraction doesn't seems to work properly i this version, I tested deleting the oot.otr file generaten on android and let the apk do the extraction 4 more times and in the first 3 tries it said the extraction was completed after less than 10 seconds, and the generated oot.otr file was less than 500kb, only the 4° attempt generated a 28,5 file which also dosen't work on PC.
In any case, I'll wait for an update, i'm really excited to see how this game performs on older devices.

**Comment by Waterdish** [comment]
Let me know if this issue still occurs with the new update (1.1.0)

**Comment by rVicio** [comment]
Yes, it works now in android 6.0.1 (galaxy J2 Prime), it managed to extract the rom and start. Tested up to the deku tree before entering the dungeon just fine, with 30 fps there's some stutters with the audio but it mantains the framerate smooth enough. There's only a small grahpical issue which I thinks it relates to the mali gpu the device has, the floor texture in the kokiri village it's not filtered, all the other floors, walls and objects are, it reminds me of the gta san andread problem in old opengl es 2.0 mali gpu based devices.
In the android 4.4.4 (grand neo plus), it still crashed with sdl2 error message, whether it's to extract
the rom or for start after copying the otr files. Tested just for curiosity, it only supports opengl es 2.0.

**Comment by Waterdish** [comment]
I'm glad to hear that it launches now. Yeah it's probably a gpu driver issue causing that, not sure I can fix that myself. Interesting to know that 4.4.4 crashes on something with SDL. Not sure what that would be about. Since Android 6 works at least, I'm going to close the issue.
